### DISCUSSION TITLE
system blew up when using a nequip pair_style in lammps

### DISCUSSION THREAD

**Discussion by Haihuichem** [mir-group/pair_nequip_allegro#77] [Q&A]
Hi, developers.
 I am currently trying to explore the possibility apply Nequip to do some liquid research, But I find that my system will always blew up, I am not sure if my settting is ok, here is my training .yaml ( the offcial test example for aspirin) and lammps .in file, 
```
units	metal
atom_style atomic
dimension 3
newton off
boundary f f f
read_data ../aspirin.data



pair_style	nequip
pair_coeff	* * ./deployed.nequip.pt2 C O H

mass 1 12
mass 2 1
mass 3 16

velocity all create 300.0 1 loop geom

neighbor 1.0 bin
neigh_modify delay 5 every 1

timestep 0.001
thermo 10

dump dump_pos all custom 200 dump.xyz id type x y z


fix  1 all nvt temp 300 300 $(100*dt)


comm_modify cutoff 7.0
compute rdfall all rdf 1000 cutoff 5.0
fix 2 all ave/time 1 2500 5000 c_rdfall[*] file si.rdf mode vector

run 5000
```

and the result:
```
Per MPI rank memory allocation (min/avg/max) = 40.34 | 40.34 | 40.34 Mbytes
   Step          Temp          E_pair         E_mol          TotEng         Press     
         0   300           -406315.56      0             -406314.78     -1421.7023    
        10   135345.56     -406058.14      0             -405708.25     -1518.8599    
        20   38730.356     -406003.74      0             -405903.61      460.94933    
        30   27325.342     -406105.89      0             -406035.25      1054.1947    
        40   16021.585     -406145.48      0             -406104.06      639.45827    
        50   8967.3536     -406153.41      0             -406130.22     -651.48886    
        60   5989.1234     -406159.78      0             -406144.3      -115.62264    
        70   5977.5772     -406152.69      0             -406137.24      1090.1945    
        80   7548.383      -406138.6       0             -406119.08     -2431.9892    
```

 
[test_repro_nequip.yaml](https://github.com/user-attachments/files/22921350/test_repro_nequip.yaml)

**Comment by cw-tan** [comment]
Hi @Haihuichem , I would not take the aspirin example too seriously as it is truly meant for just being able to run our CI tests. 
Note that in the training script, you are training on 5 frames.
```yaml
  split_dataset:
    file_path: ${dataset_file_name}
    train: 5
    val: 5
    test: 5
```
You might consider some benchmark datasets here: https://nequip.readthedocs.io/en/latest/api/datamodule.html#nequip.data.datamodule.WaterDataModule to be more useful if you want to run production MD.  

Stability issues on production quality datasets can typically be resolved with a combination of 

- using ZBL: https://nequip.readthedocs.io/en/latest/guide/configuration/model.html#ziegler-biersack-littmark-zbl-potential
- tuning your weight decay in AdamW carefully (roughly speaking, larger weight decay makes the potential energy surface smoother and prevents unphysical oscillatory behavior that arises when the model overfits to the data, but if it's too large, you impede learning)
- inspect dimer curves, e.g. with https://github.com/mir-group/nequip/blob/main/misc/plot_dimers.py. While it only reveals two-body interaction behavior, you can get a qualitative sense of the smoothness of your potential and potentially see other obvious problems as you try to iterate on refining your training dataset and hyperparameter tuning.

**Reply by Haihuichem** [reply]
Many thanks, it really works for me, actually I also use a total 2000 frames(2ns) AIMD reference data for DCM. Although the energy is more stable now, the model still fail to capture the correct topology. I will try to retune  my model according to your advice. Btw, the original paper of Nequip(Nat Commun 13, 2453 (2022)) mentioned that you compare the performance with deepMD both in water and ice and said will do dynamical research in the future, may I know if your system also meets such a problem or run smoothly?
